collect/collect.go

```diff
@@ -376,18 +378,22 @@ func (i *InMemCollector) collect() {
 					return
 				}
 				i.processSpan(sp)
+				i.Metrics.Gauge("collector_collect_loop_duration_ms", float64(time.Now().Sub(startTime).Milliseconds()))
```

Comment:
I'm not sure I understand why the "collector_collect_loop_duration_ms" metric is being recorded when handling incoming spans, peer spans and at the end of the for outer for loop.

I expected it only happen at the end of the outer for loop.

Reply:
It's because of the `continue` statements. I have removed them since they are unnecessary to have right now